pkg/adt/interval_tree.go

```diff
@@ -472,8 +472,16 @@ func (ivt *intervalTree) Insert(ivl Interval, val any) {
 	x := ivt.root
 	for x != ivt.sentinel {
 		y = x
-		if z.iv.Ivl.Begin.Compare(x.iv.Ivl.Begin) < 0 {
+		// Split on left endpoint. If left endpoints match, instead split on right endpoint.
```

Comment:
Can you update the function docstring?

Reply:
Sorry, missed this comment.  The comment for the Insert method isn't accurate anymore, we need to update it. We also need to add comment for the `find` method. @redwrasse 

https://github.com/etcd-io/etcd/blob/c849507a3895a93742d83845635e17a069a8ee2e/pkg/adt/interval_tree.go#L437-L468

Reply:
@serathius @ahrtr I'll open an MR for updating the comments.

Reply:
Created a pull request with docstring updates: https://github.com/etcd-io/etcd/pull/20015